### PR TITLE
Update readme and error handling

### PR DESCRIPTION
**Pull request by shomix**
Readme: https://github.com/spheronFdn/sdk/tree/update-readme-md/packages/storage
Also fixed the error handling, the parsing was wrong.

### CODE REVIEW

**Review by aayushmahapatra — APPROVED** [review]
lgtm

File: packages/storage/src/upload-manager/index.ts [hunk 1]
```diff
@@ -91,7 +91,7 @@ class UploadManager {
       );
       return response.data;
     } catch (error) {
-      const errorMessage = error?.data?.message || error?.message;
+      const errorMessage = error?.response.data?.message || error?.message;
```
Comment (rekpero) [inline packages/storage/src/upload-manager/index.ts:94]:
```suggestion
      const errorMessage = error?.response?.data?.message || error?.message;
```

File: README.md [hunk 2]
```diff
@@ -43,26 +47,42 @@ import SpheronClient, { ProtocolEnum } from "@spheron/storage";
...
-const { uploadId, bucketId, protocolLink, dynamicLinks } = await client.upload(filePath, { protocol: ProtocolEnum.IPFS, name });
+let currentlyUploaded = 0;
+const { uploadId, bucketId, protocolLink, dynamicLinks } =
+      await client.upload(filePath, {
+        protocol: ProtocolEnum.IPFS,
+        name,
+        onUploadInitiated: (uploadId) => {
+          console.log(`Upload with id ${uploadId} started...`);
+        },
+        onChunkUploaded(uploadedSize, totalSize) {
```
Comment (rekpero) [inline README.md:58]: why is this different looking than `onUploadInitiated` might confuse beginner devs

**Review by rekpero — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (shomix) [reply README.md:58]: Ahh, didn't notice it. Fixing it.

Comment (shomix) [reply README.md]: Fixed.

File: packages/storage/src/upload-manager/index.ts (hunk 1 above)
Comment (shomix) [reply packages/storage/src/upload-manager/index.ts]: Fixed.

File: packages/storage/README.md [hunk 3]
```diff
@@ -36,25 +36,39 @@ import SpheronClient, { ProtocolEnum } from "@spheron/storage";
 
 ...
 const client = new SpheronClient({ token });
-const { uploadId, bucketId, protocolLink, dynamicLinks } = await client.upload(filePath, { protocol: ProtocolEnum.IPFS, name });
+let currentlyUploaded = 0;
```
Comment (rekpero) [inline packages/storage/README.md:39]: update the npm version in the readme npm badge

File: packages/storage/README.md [hunk 4]
```diff
@@ -36,25 +36,39 @@ import SpheronClient, { ProtocolEnum } from "@spheron/storage";
 
 ...
```
Comment (rekpero) [inline packages/storage/README.md:37]: remove these dots

File: packages/storage/README.md [hunk 5]
```diff
@@ -36,25 +36,39 @@ import SpheronClient, { ProtocolEnum } from "@spheron/storage";
...
+        name,
+        onUploadInitiated: (uploadId) => {
+          console.log(`Upload with id ${uploadId} started...`);
+        },
+        onChunkUploaded: (uploadedSize, totalSize) => {
+          currentlyUploaded+=uploadedSize;
+          console.log(`Uploaded ${currentlyUploaded} of ${totalSize} Bytes.`);
+        },
+      });
 ...
```
Comment (rekpero) [inline packages/storage/README.md:52]: same here

File: README.md [hunk 6]
```diff
@@ -43,26 +47,42 @@ import SpheronClient, { ProtocolEnum } from "@spheron/storage";
...
+        name,
+        onUploadInitiated: (uploadId) => {
+          console.log(`Upload with id ${uploadId} started...`);
+        },
+        onChunkUploaded: (uploadedSize, totalSize) => {
+          currentlyUploaded+=uploadedSize;
+          console.log(`Uploaded ${currentlyUploaded} of ${totalSize} Bytes.`);
+        },
+      });
 ...
```
Comment (rekpero) [inline README.md:63]: same here

File: README.md [hunk 7]
```diff
@@ -43,26 +47,42 @@ import SpheronClient, { ProtocolEnum } from "@spheron/storage";
 
 ...
```
Comment (rekpero) [inline README.md:48]: same here

**Review by rekpero — CHANGES_REQUESTED** [review]
some more RC

File: packages/storage/README.md (hunk 3 above)
Comment (shomix) [reply packages/storage/README.md:38]: Updated to v8.0.5 ( this is the one I use ).
If some specific version is needed, please say which one

File: packages/storage/README.md (hunk 4 above)
Comment (shomix) [reply packages/storage/README.md]: Removed

File: packages/storage/README.md (hunk 5 above)
Comment (shomix) [reply packages/storage/README.md]: Removed

File: README.md (hunk 6 above)
Comment (shomix) [reply README.md]: Removed

File: README.md (hunk 7 above)
Comment (shomix) [reply README.md]: Removed

File: packages/storage/README.md [hunk 8]
```diff
@@ -14,7 +14,7 @@
 
 <p align="center">  
   <a href="https://www.npmjs.com/package/@spheron/storage" target="_blank" rel="noreferrer">
-    <img src="https://img.shields.io/static/v1?label=npm&message=v1.0.3&color=green" />
+    <img src="https://img.shields.io/static/v1?label=npm&message=v8.0.5&color=green" />
```
Comment (rekpero) [inline packages/storage/README.md:17]: not 8.0.5, this is the version of the storage package not node package

**Review by rekpero — CHANGES_REQUESTED** [review]

File: packages/storage/README.md (hunk 8 above)
Comment (vocoront1) [reply packages/storage/README.md]: haha :D didn't know about this. Fixed it.

**Review by rekpero — APPROVED** [review]

**Review by rekpero — APPROVED** [review]

**Review by dusanstanis — APPROVED** [review]
lgtm !

**Pull request merged**